src/credentials/tests/TestChipCert.cpp

```diff
@@ -0,0 +1,635 @@
+/*
```

Comment:
I did not carefully review the test...  It might be good to add tests for various things mentioned in the review, though, especially around invalid certificates.  In fact, I'm not seeing much testing of invalid certificates here, either invalid TLV or invalid ASN1.

Reply:
https://github.com/project-chip/connectedhomeip/issues/4324